### ISSUE TITLE
cannot get ctrlp_user_command to work.

### ISSUE THREAD

**Issue by SrinivasDJ** [ctrlpvim/ctrlp.vim#129]
I tried setting ag with ctrlp as per documentation but I get "NO ENTRIES" as the output. To test further I removed all other plugins and with clean vimrc, I tried: let g:ctrlp_user_command = 'find %s -type f'. Even this doesn't work and I get empty result. Am I missing any config? My vim is from debian with version 7.4 on linux mint. Modified ctrlp_match_func with pymatched works btw.

--

**Comment by mattn** [comment]
if just run ag in terminal, it works correctly?

**Comment by SrinivasDJ** [comment]
Yes, ag runs correctly in terminal. As mentioned if I assign 'find %s -type f' to user_command, that too returns "No Entries". Please let me know if I can gather any useful/debug information. Am using the latest ctrlp plugin.
--
Thanks.

**Comment by ludovicchabant** [comment]
Do you get "no entries" right from the beginning? (i.e before typing any search terms)
If you type `:echo printf(g:ctrlp_user_command, '')`, do you get `find  -type f`?
The only thing I can think of is that maybe CtrlP is in the wrong current directory... I think you can check where CtrlP is running the search from by hitting `<C-P>` and looking at the right of the prompt window.

**Comment by emilford** [comment]
I also get "No Entries" when using `ag`, though some times if I ctrl-c out of the search and retry a few times I'll get entries. The `ctrlp_user_command` I have set up works every time when run from the command line and as best as I can tell (based on the comment above) the search is being run from the correct directory. Whether or not "No Entries" is returned is frequent but also sporadic. It's been months and I'm still unclear what causes the difference in behavior.

**Comment by emilford** [comment]
@SrinivasDJ Have you had any luck resolving this issue?

**Comment by SrinivasDJ** [comment]
@emilford, I had stopped using ctrlp for sometime now. Last night I downloaded the lastest ctrlp and now am able to use it with Ag. Here are my versions: vi: 7.4, Ag: 0.26.0.1.
The only difference between now and when I saw the problem was the in versions of ag and vim (Changed distro from Mint to Debian 8.2). My vimrc and all other plugins are on same versions.

**Comment by emilford** [comment]
@SrinivasDJ Thanks for the response. 

I was trying a few things the other morning and found that changing lines in my `~/.agignore` that referenced nested directories to include a leading forward slashe resolved the "No Entries" for me.

e.g., `app/assets/fonts` -> `/app/assets/fonts`

I have not had time to look into why this is or whether it's a bug in (likely) `Ag` or was simply an issue in my configuration, but perhaps it will help out others facing this issue.
